src/modules/fancyzones/editor/FancyZonesEditor/CanvasZone.xaml.cs

```diff
@@ -16,192 +17,336 @@ namespace FancyZonesEditor
     /// </summary>
     public partial class CanvasZone : UserControl
     {
-        public CanvasLayoutModel Model { get; set; }
-
-        public int ZoneIndex { get; set; }
+        public CanvasZone()
+        {
+            InitializeComponent();
+            Canvas.SetZIndex(this, zIndex++);
+        }
 
-        private readonly Settings _settings = ((App)Application.Current).ZoneSettings;
+        private CanvasLayoutModel model;
 
-        private static readonly int _minZoneWidth = 64;
-        private static readonly int _minZoneHeight = 72;
-        private static int _zIndex = 0;
+        private int zoneIndex;
 
-        public CanvasZone()
+        public enum ResizeMode
         {
-            InitializeComponent();
-            Panel.SetZIndex(this, _zIndex++);
+            ModeLow = 1,
+            ModeHigh = 2,
+            ModeBoth = 3,
```

Comment:
Since the enum already states that this is the ResizeMode, I would suggest to rename the values to:
 - BottomEdge
 - TopEdge
 - BothEdges

Reply:
Perhaps LowEdge, HighEdge, BothEdges?

Reply:
Top and Bottom are better terms in this case.

Reply:
Or are these referring to the Y axis as well?

Reply:
They are used in both X and Y axes.

Reply:
In case of Y, top edge is the left edge, correct?